### ISSUE TITLE
Path "../../../../../../lib/web/mage/gallery/gallery.less" cannot be used with directory

### ISSUE THREAD

**Issue by iget-esoares** [magento/magento2#20815]
### Summary (*)
My theme extends the gallery.css with this file:

> app/design/frontend/Vendor/themename/web/mage/gallery/gallery.less

After updating from 2.2.6 to 2.2.7, when trying to do `grunt exec:themename` causes the following error:

> >> 
> >> 
> >> [Magento\Framework\Exception\ValidatorException]                                                                                                       
> >>   Path "../../../../../../lib/web/mage/gallery/gallery.less" cannot be used with directory "/var/www/app/design/frontend/Vendor/themename/web/i18n/pt_BR/"  
> >>                                                                                                                                                          
> >> 
> >> dev:source-theme:deploy [--type TYPE] [--locale LOCALE] [--area AREA] [--theme THEME] [--] [<file>]...
> >> Exited with code: 1.
> >> Error executing child process: Error: Process exited with code 1.
> Warning: Task "exec:themename" failed. Use --force to continue.
> 

### Examples (*)

Described above.

My gallery.less file contains this:

```
@import '../../../../../../../../lib/web/mage/gallery/gallery';

.fotorama__caption {
  display: none;
}
```

**Comment by magento-engcom-team** [comment]
Hi @iget-esoares. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@iget-esoares do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @awviraj. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @rabhiaziz I'm not able to reproduce following steps you described.
The GitHub issue tracker is intended for Magento Core technical issues only.
Please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/) site for advice or general discussion about this issue.

**Comment by iget-esoares** [comment]
@engcom-backlog-nazar That's not an advice or general discussion. It used to work on previous version 1.2.6 and do not work after update to 1.2.7. Nothing else changed, just the magento version.

Also there's no mention on changelog for any breaking change that cause this, I spent a hour reading it.

Really hate when you guys just close issues this way.

**Comment by ghost** [comment]
@iget-esoares what version ? 1.2.6 ? maybe 2.2.6 ? i'm found another issue that contain more clear steps https://github.com/magento/magento2/issues/20857 so this may be related to yours

**Comment by ghost** [comment]
@iget-esoares also this error because you have front real Path**"**../../../../../../**lib/web/mage/gallery/gallery.less" cannot be used with directory** 
you need to `Path "/var/www/lib/web/mage/gallery/gallery.less" cannot be used with directory "/var/www/app/design/frontend/Vendor/themename/web/i18n/pt_BR/"`

**Comment by dverkade** [comment]
Hi,

Please have a look at magento/magento2#22627. I was able to reproduce this issue and fix it by changing the import file in the less.file.

**Comment by engcom-Charlie** [comment]
Hello @iget-esoares 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates 2.3-develop branch by provided steps.

We are closing this issue due to branch 2.2-develop was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3.
See Accepted pull requests and ported code for more details

If you still faced this issue on 2.3 please create a new Issue with all required details according to Issue reporting guidelines

Thanks for your report!
